### PR TITLE
Priorityq

### PR DESCRIPTION
**Pull request by CCallahanIV**
TODO:
 - Clean up tests
 - Update README
 - Check boundary conditions for input

### CODE REVIEW

File: src/test_priorityq.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    """The fixture creates a filled priority queue."""
+    from priorityq import PriorityQ
+    new_pq = PriorityQ(TEST_SET)
+    return new_pq
+
+
+def test_creation_of_empty_priority_q(empty_priority_q):
+    """The creates an empty queue and tests the size."""
+    assert len(empty_priority_q) == 0
+    assert len(empty_priority_q._high_p) == 0
```
Comment (Bl41r) [inline src/test_priorityq.py:27]: Are you sure about what you think this len is?

File: src/priorityq.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            val = self._pdict[self._high_p].dequeue()
+            if len(self._pdict[self._high_p]) == 0:
+                del self._pdict[self._high_p]
+                try:
+                    self._high_p = min(self._pdict.keys())
+                except ValueError:
+                    self._high_p = None
+            self._size -= 1
+            return val
+        except IndexError:
```
Comment (Bl41r) [inline src/priorityq.py:46]: Are you sure this is what actually happens?

File: src/priorityq.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            raise IndexError("Cannot pop from empty Priority Q.")
+
+    def peek(self):
+        """Return the value of the item with greatest priority, do not remove it."""
+        return self._pdict[self._high_p].peek()
```
Comment (Bl41r) [inline src/priorityq.py:51]: What happens when the pqueue is empty here?

File: src/priorityq.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
+"""This module contains an implementation of the PriorityQ."""
```
Comment (Bl41r) [inline src/priorityq.py:1]: More explanation about how to use this module would be nice here...how priority is scored, etc...
